src/domain/templates/templates.types.ts

```diff
@@ -4,21 +4,40 @@ const baseFields = z.object({
   content: z.string(),
 });
 
-const $BaseTemplateSchema = z.object({
+const $BaseTemplateInput = z.object({
   name: z.string(),
   type: z.enum(['NHS_APP', 'SMS', 'EMAIL', 'LETTER']),
   version: z.number(),
   fields: baseFields,
 });
 
-const $NHSAppTemplateSchema = $BaseTemplateSchema.extend({
+const $NHSAppTemplateInput = $BaseTemplateInput.extend({
   type: z.literal('NHS_APP'),
 });
 
-export const $TemplateSchema = z.discriminatedUnion('type', [
-  $NHSAppTemplateSchema,
+const $EmailTemplateInput = $BaseTemplateInput.extend({
+  type: z.literal('EMAIL'),
+});
+
+const $SMSTemplateInput = $BaseTemplateInput.extend({
+  type: z.literal('SMS'),
+});
+
+const $LetterTemplateInput = $BaseTemplateInput.extend({
+  type: z.literal('LETTER'),
+});
+
+export const $TemplateInput = z.discriminatedUnion('type', [
+  $NHSAppTemplateInput,
+  $EmailTemplateInput,
+  $SMSTemplateInput,
+  $LetterTemplateInput,
 ]);
 
-export type Template = z.infer<typeof $BaseTemplateSchema>;
+export const $Template = $TemplateInput.and(z.object({ id: z.string() }));
+
+export type TemplateInput = z.infer<typeof $TemplateInput>;
+export type Template = z.infer<typeof $Template>;
```

Comment:
I don't think we need `$Template` just for typings. We can extending this type like;

```
export type Template = z.infer<typeof $TemplateInput> & { id: string }
```

Reply:
this was unused at the time of the comment but is now used in relation to your other comment